### DISCUSSION TITLE
How to update a plot in a loop?

### DISCUSSION THREAD

**Discussion by AJAnderson** [emilk/egui#666] [Q&A]
Hey guys,

Just getting to grips with egui. The example in the docs has line being created and then the plot widget being added to the ui https://docs.rs/egui/0.14.0/egui/widgets/plot/struct.Plot.html.

In the reference template, the app stuff is outside the main.rs file and is called into using eframe::run_native.

I have a data parsing library I'm working on where I'd like to be able to load and plot graphs. Should I be doing that in my main.rs somehow or in the app somehow. I'm a little confused as to exactly what happens with eframe::run_native. Is a new thread being spun up? Should my app.rs be calling out to my library directly, and the main only exists to kick off the gui app?

Thanks in advance

**Comment by emilk** [comment]
`run_native` won't return until your app is done. All your app logic should go in `App::update`. So for instance, you can add an "Open" button to open a file, then parse it either directly (if it is small) or in a separate thread (if it is large), and then store it in your app and show it.

For an example you may want to look at https://github.com/EmbarkStudios/puffin/blob/main/puffin_viewer/src/main.rs

**Reply by AJAnderson** [reply]
I  can't mark this as answer (button is non-responsive) but thank you!

**Reply by emilk** [reply]
If you run `puffin_viewer` (`cargo install puffin_viewer && puffin_viewer`) does the open/save buttons work for you? If yes, then you should be able to do the same thing, if no there is a bug that should be fixed!

**Reply by AJAnderson** [reply]
I haven't tried running puffin viewer but I did copy the methodology to my own code and it did work!

Now I'm trying to figure out why the side panel can't be resized beyond a certain width?

edit: I know it's possible in the template, but somehow I'm not able to shrink the central panel below a certain amount. In my code all it has is a heading.

edit2: It seems to be tied to the maximum size of text in the side panel. As far I can understand the addition of the bottom up, centered URL is what allows the side panel to be arbitrarily resizable. If it's deleted then it's driven by the text entry box. This is a little unintuitive.

**Reply by AJAnderson** [reply]
@emilk sorry to ping, but any guidance on the "right way" to enable or disable arbitrary resize?

**Reply by emilk** [reply]
You can add `ui.allocate_space(ui.available_size());` to the end of a panel or window, which will just add empty space as needed.

Also see the `min_width/max_width/min_height/max_height` methods on the panels.
